Gem/Code/Source/Components/Multiplayer/EnergyBallComponent.cpp

```diff
@@ -275,10 +280,9 @@ namespace MultiplayerSample
         m_hitEvent.m_shooterNetEntityId = m_shooterNetEntityId;
         m_hitEvent.m_projectileNetEntityId = GetNetEntityId();
 
-        AzPhysics::SimulatedBodyComponentRequestsBus::Event(GetEntityId(), &AzPhysics::SimulatedBodyComponentRequestsBus::Events::DisablePhysics);
-        Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequestBus::Events::SetLinearVelocity, AZ::Vector3::CreateZero());
-
         RPC_BallExplosion(m_hitEvent);
+
+        Multiplayer::GetNetworkEntityManager()->MarkForRemoval(GetNetBindComponent()->GetEntityHandle());
```

Comment:
Kill event is safe since you've guarded it with the ball active check, yup it's safe.. it's how we handle pretty much all network spawnables except for the gems, and yeah, it did affect running the explosion effect, so I had to add a delay timer on the actual entity destruction.. Good call on that.